homeassistant/components/modbus/__init__.py

```diff
@@ -273,6 +277,10 @@
             vol.Optional(CONF_TEMPERATURE_UNIT, default=DEFAULT_TEMP_UNIT): cv.string,
             vol.Exclusive(CONF_HVAC_ONOFF_COIL, "hvac_onoff_type"): cv.positive_int,
             vol.Exclusive(CONF_HVAC_ONOFF_REGISTER, "hvac_onoff_type"): cv.positive_int,
+            vol.Inclusive(CONF_CURRENT_TEMP_SCALE, "current_temp"): vol.Coerce(float),
```

Comment:
Why is it not allowed just to add scale or offset, why does it need to be both.

If I remember correctly, you only needed scale but someone else asked for offset, so it is clear that some devices do not need both scale and offset.

Reply:
My position is that if someone tells me how to do it correctly, I will do it faster than sending 30 variants that will be rejected. And you will save time by not having to check incorrect solutions.